measures/UtilityBillCalculations/measure.rb

```diff
@@ -0,0 +1,351 @@
+# see the URL below for information on how to write OpenStudio measures
+# http://nrel.github.io/OpenStudio-user-documentation/reference/measure_writing_guide/
+
+require 'erb'
+require 'csv'
+require 'matrix'
+
+#start the measure
+class UtilityBillCalculations < OpenStudio::Measure::ReportingMeasure
+
+  # human readable name
+  def name
+    return "Utility Bill Calculations"
+  end
+
+  # human readable description
+  def description
+    return "Calls SAM SDK."
+  end
+
+  # human readable description of modeling approach
+  def modeler_description
+    return "Calls SAM SDK."
+  end 
+  
+  # define the arguments that the user will input
+  def arguments()
+    args = OpenStudio::Measure::OSArgumentVector.new
+    
+    arg = OpenStudio::Measure::OSArgument.makeStringArgument("run_dir", true)
+    arg.setDisplayName("Run Directory")
+    arg.setDescription("Relative path of the run directory.")
+    arg.setDefaultValue("..")
+    args << arg    
+    
+    arg = OpenStudio::Measure::OSArgument::makeStringArgument("api_key", false)
+    arg.setDisplayName("EIA API Key")
+    arg.setDescription("Call the API and find EIA ID(s) matching the epw.")
+    args << arg
+    
+    arg = OpenStudio::Measure::OSArgument::makeStringArgument("json_file_path", false)
+    arg.setDisplayName("JSON File Path")
+    arg.setDescription("Provide this instead of calling the API.")
+    args << arg
+    
+    arg = OpenStudio::Measure::OSArgument::makeDoubleArgument("analysis_period", false)
+    arg.setDisplayName("Analysis Period")
+    arg.setUnits("yrs")
+    arg.setDefaultValue(1)
+    args << arg
+    
+    return args
+  end
+  
+  def outputs
+    result = OpenStudio::Measure::OSOutputVector.new
+    result << OpenStudio::Measure::OSOutput.makeStringOutput("grid_cells")
+    result << OpenStudio::Measure::OSOutput.makeStringOutput("total_site_electricity")
+    buildstock_outputs = [
+                          "total_site_natural_gas",
+                          "total_site_propane",
+                          "total_site_other"
```

Comment:
I suggest removing the word "site" from the above 4 strings, as site/source doesn't have any meaning for utility bills.